### ISSUE TITLE
resetProgress doesn't work until the app is restarted

### ISSUE THREAD

**Issue by jpaas** [danielniemeyer/DNTutorial#2]
DNTutorial.resetProgress() does work, but not until the tutorial dictionary is reloaded from NSUserDefaults. That doesn't happen until the singleton shared instance is recreated. I was hoping to have a button that allows the user to show the tutorial again if they want to, but currently I can't find a way to make that work.

**Comment by danielniemeyer** [comment]
Have you tried presenting your tutorial steps on a different place than viewDidLoad or viewDidAppear?

**Comment by jpaas** [comment]
Yes actually. I present my tutorial when a modal view controller finishes.

**Comment by jpaas** [comment]
If I reinitialize the userDefaults property on the sharedInstance, resetProgress works properly.

```objc
+ (void)resetProgress;
{
    // Restore to factory settings
    [[NSUserDefaults standardUserDefaults] removeObjectForKey:sUserDefaultsKey];
    [[NSUserDefaults standardUserDefaults] synchronize];
    [self sharedInstance].userDefaults = [DNTutorialDictionary dictionary];
}
```

**Comment by danielniemeyer** [comment]
Actually have you tried playing with it recently in release 0.1.4? It fixes a bug with how reset was previously handled.

**Comment by jpaas** [comment]
I just tried it with 0.1.4. It doesn't work without that extra line of code.

**Comment by danielniemeyer** [comment]
I cleaned up that a bit and included it with release 0.1.5, let me know if you still experience issues with it!

**Comment by jpaas** [comment]
I tried 0.1.6 and it works now. Thanks!
